prow/repoowners/repoowners.go

```diff
@@ -338,6 +347,9 @@ func (o *RepoOwners) applyConfigToPath(path string, config *ownersConfig) {
 	if len(config.Labels) > 0 {
 		o.labels[path] = sets.NewString(config.Labels...)
 	}
+	if config.Options != defaultDirOptions {
```

Comment:
fixing the unit test made me realize that I was also storing a `dirOptions` for every dir even if they were unset, hence this. :shrug: